app/models/concerns/audit_extensions.rb

```diff
@@ -93,15 +86,42 @@ def known_auditable_types
       end
 
       def taxed_only_by_location
-        locations = Location.current || Location.my_locations.reorder(nil)
-        by_auditable_types(location_taxable).
-          where(id: TaxableTaxonomy.where(taxonomy: locations, taxable_type: 'Audited::Audit').select(:taxable_id))
+        join_user_locations.where(arel_taxed_only_by_location)
       end
 
       def taxed_only_by_organization
-        organizations = Organization.current || Organization.my_organizations.reorder(nil)
-        by_auditable_types(organization_taxable).
-          where(id: TaxableTaxonomy.where(taxonomy: organizations, taxable_type: 'Audited::Audit').select(:taxable_id))
+        join_user_organizations.where(arel_taxed_only_by_organization)
+      end
+
+      def taxed_and_untaxed
+        clauses = []
+        unless !Organization.current && User.current.admin?
```

Comment:
I think flipping around the conditional will make it easier to understand (also for location below):
```suggestion
        if Organization.current || !User.current.admin?
```

Reply:
Done